### ISSUE TITLE
Removal of :focus styles in tutorial code

### ISSUE THREAD

**Issue by colabottles** [facebook/react#19196]
In the tutorial, the `:focus` styles are removed, which as we all know _(or should)_ that this is not the best way to do this because of accessibility reasons. I know it's a tutorial for a Tic-Tac-Toe game, but this teaches bad accessibility practices and needs to be, as well as should be fixed.

Keyboard accessibility is crucial to some, and using the **TAB** key to navigate is also crucial to those folks as well. As well as others who use the keyboard for navigation and/or a screen reader.

Whether you remove the code entire, which I have done and it is fine with thin dotted borders in Firefox, or the glowing green border color in Chrome/Edge and Safari, this sets a bad example. You could change this code below:

```
.square:focus {
  outline: none;
}
```

to look the same across the major modern browsers using:

```
.square:focus {
  outline: thin dotted;
}
```

It is my experience in the accessibility and programming spaces, that if you foster the idea of accessibility at the most fundamental levels, like tutorials, this fosters a pattern of positivity towards accessibility. Teaching juniors to remove the default`:focus` styling in the browser needs to stop so they can learn the fundamentals of accessibility as well as programming.

**Comment by Lewiscowles1986** [comment]
I'll just leave this here https://github.com/facebook/react/search?l=CSS&q=%3Afocus

**Comment by bvaughn** [comment]
@colabottles The tutorial (and the rest of the React website) has its own repo, with a slightly different set of contributors. Please file this there: https://github.com/reactjs/reactjs.org

Not sure I understand the purpose of the comment by @Lewiscowles1986. DevTools uses custom focus styles in several places. If you look at the individual search results, you'll see that color/background-color/other changes are made instead of the focus ring.

**Comment by Lewiscowles1986** [comment]
@bvaughn if you look at the link I posted, you are also doing that in this repo in several CSS files, which might be generated from emotion or some JS, but STILL have the issue.

> focus visible is also about to become a single A criterion because focus visible enhanced is coming in 2.2 as AA.
> never unset or reset the user agent default focus indication – without specifically replacing it with a stronger indicator.
> when describing this, i recommend more urgency and importance than simply saying it is bad practice.
> it creates a barrier. doing so is against the law in over 17 countries / regions with a strong advisement in over 75 countries – as access is a human right.
> - Charles Hall, specifically on a thread about this issue

**Comment by bvaughn** [comment]
@Lewiscowles1986 Obviously I looked at the link you sent before commenting. I opened the first several files to confirm that I wasn't just hiding focus style, but *customizing* it (replacing the standard focus UI with a custom one). This should be okay, given the quote you yourself just provided:
> never unset or reset the user agent default focus indication – **without specifically replacing it with a stronger indicator.**

**Comment by Lewiscowles1986** [comment]
your indicator is none.

>  outline: none;

That isn't **specifically replacing it with a stronger indicator.**

Do you have an example of something you are inferring here?

**Comment by bvaughn** [comment]
Look at the CSS files, Lewis. When I set "indicator" to none- I change color, background-color, or style of a descendant tag (depending on the UI). re-read my above comments. If you still disagree, point to a specifc file.

**Comment by Lewiscowles1986** [comment]
I disagree with every case you've just raised. I also checked and then changed my search query from focus to outline to try to get broader focus. I am not claiming to be a CSS parser so I may have missed something. 

I don't see anything stronger than the outline, or any instance of strengthened outine.

It's visually inconsistent with the way the rest of the web works out of the box.

That said I'd be interested to hear why you felt the other properties are as good or better for accessibility. I notice what you are saying in some cases, although not all, which could be a search-UI problem. Text color change is the one I struggle with most.

If you are saying you've considered the above, while I've no right and am not asserting a right to free consultancy, I'll not continue some internet argument. It does seem another example of React going it's own way inexplicably. Perhaps you wrote a blog you could link to about why instead of outline you picked three totally different UI changes?

**Comment by bvaughn** [comment]
I've asked a few times for you to point out a specific example, so we could talk about it rather than generalities. "I disagree with everything you said" is not productive and I don't really know how to continue the discussion, so I'm going to bow out of this thread after one final reply.

I will post [your own quote](https://github.com/facebook/react/issues/19196#issuecomment-650008440) back to you one more time:

> focus visible is also about to become a single A criterion because focus visible enhanced is coming in 2.2 as AA.
> never unset or reset the user agent default focus indication – without specifically replacing it with a stronger indicator.
> when describing this, i recommend more urgency and importance than simply saying it is bad practice.
> it creates a barrier. doing so is against the law in over 17 countries / regions with a strong advisement in over 75 countries – as access is a human right.
> - Charles Hall, specifically on a thread about this issue

The outline rects I removed *were* replaced "with a stronger indicator". The reason they were removed was often because the glow clipped in a way such that it was only partially visible, so I used background color (really brightness, not color) as an alternate indicator.

**Comment by Lewiscowles1986** [comment]
I said everything and pointed out two examples

* background color
* text color

I'll be honest Bow out, enjoy your weekend and life as far as I'm concerned this is peak React Dev. If something is clipping outline, surely removing the clipping, rather than coming up with hot takes on alternatives 🙄
